Little improve code organization.

polina-c opened the pull request:
Resolve cycle around local storage.

devoncarew reviewed (commented):
breaking the cycle (local_interface.dart <==> local_interface/web.dart, ...) looks good

I'm worried about the several `DEPENDENCIES.md` files; these will get out of date, and or, local devs will need to install before contributing.

I would either remove from this PR (and .gitignore them), or, remove + ignore, plus set up a github workflow to run the dep tool and fail on any cycles discovered; thoughts?

polina-c commented:
Sounds good!

Removed diagrams by adding them to gitignore.

Also moved some code to the folder 'primitives.

Will add validation for loops in other PR.

polina-c commented:
![Screenshot 2025-04-01 at 5 14 25 PM](https://github.com/user-attachments/assets/3fdd2966-ff96-4a46-83fd-b71666fc88b2)

polina-c commented:
![Screenshot 2025-04-02 at 7 47 26 AM](https://github.com/user-attachments/assets/b37f2179-8652-480d-856c-f4f0be786ec1)

devoncarew reviewed (commented):
I like the removal of the dep loop for the local storage related files. I'm less certain about the new code organization - using `primitives/` - for everything that doesn't currently have a dep outside that folder. I'm not sure we want to structure our code strictly around that aspect. DartPad is s small enough codebase that having mostly a flat list of files in `lib/` works, w/ some grouping for topics (`editor/`, ...).

@johnpryan - thoughts?

polina-c commented:
Reverted creation of 'primitives' to address it separately.

devoncarew reviewed (approved):
🎉
